lib/elixir/lib/module/checker.ex

```diff
@@ -104,6 +109,17 @@ defmodule Module.Checker do
     ) == []
   end
 
+  defp deprecated_reason(module, fun, arity) do
+    with {^module, binary, _path} <- :code.get_object_code(module),
+         {:ok, {^module, [{'ExDp', chunk}]}} <- :beam_lib.chunks(binary, ['ExDp']),
+         {:elixir_deprecated_v1, deprecated} <- :erlang.binary_to_term(chunk),
+         {_key, reason} <- List.keyfind(deprecated, {fun, arity}, 0) do
+      reason
+    else
+      _ -> nil
+    end
+  end
```

Comment:
Should we have some kind of a cache for this? Hitting disk for every single function call seems super expensive.

Reply:
Yes, we will have a cache for this. 👍